game-core/src/main/java/games/strategy/engine/history/change/units/RemoveUnits.java

```diff
@@ -0,0 +1,136 @@
+package games.strategy.engine.history.change.units;
+
+import games.strategy.engine.data.Change;
+import games.strategy.engine.data.CompositeChange;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.changefactory.ChangeFactory;
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.engine.history.change.HistoryChange;
+import games.strategy.triplea.formatter.MyFormatter;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import lombok.AccessLevel;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.experimental.FieldDefaults;
+import org.apache.commons.text.StringSubstitutor;
+import org.triplea.java.collections.IntegerMap;
+
+/**
+ * Removes a set of units in a location and adds a history event
+ *
+ * <p>Transforms units to other units if needed. See {@link TransformDamagedUnits}
+ */
+@FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+@Getter
+@EqualsAndHashCode
+public class RemoveUnits implements HistoryChange {
```

Comment:
I do wonder if these objects should be suffixed with `HistoryChange` so that they are not quite so context dependent. WDYT @trevan ?

Reply:
Sounds good.  I've renamed them @DanVanAtta